src/multidim/setdiff.jl

```diff
@@ -44,65 +29,21 @@ Algorithm: Start from the total overlap (in all directions);
 expand each direction in turn.
 """
 function setdiff(A::IntervalBox{N,T}, B::IntervalBox{N,T}) where {N,T}
-    X = [labelled_setdiff(a, b) for (a, b) in zip(A.v, B.v)]
-    # ordered such that the first in each is the excluded interval
-
-    first = [ i[1] for i in X ]
-    labels = [i[2] for i in first]
-
-    any(labels .== -1) && return [A]  # no overlap
-
-    # @assert all(labels .== 1)
-
-    excluded = [i[1] for i in first]
 
-    result_list = IntervalBox{N,T}[]
-
-    for dimension in N:-1:1
-        for which in X[dimension][2:end]
-            excluded[dimension] = which[1]
-            push!(result_list,
-                IntervalBox(excluded[1:dimension]..., A[dimension+1:N]...))
+    intersection = A ∩ B
+    isempty(intersection) && return [A]
+
+    result_list = fill(IntervalBox(emptyinterval(T), N), 2*N)
+    offset = 0
+    x = Array(A.v)
+    @inbounds for i = 1:N
+        tmp = _setdiff(A[i], B[i])
+        @inbounds for j = 1:2
+            x[i] = tmp[j]
```

Comment:
It may be better (more efficient) to not have `x` be an array and just use `setindex` on an `SVector` for this.

Reply:
this helped! it reduced the allocations 2 -> 1 and the execution time is now 75 ns!

```
julia> @btime setdiff($X, $Y)
  76.495 ns (1 allocation: 368 bytes)
6-element Vector{IntervalBox{3, Float64}}:
 [0, 0.5] × [3, 4] × [5, 6]
 [0.6, 2] × [3, 4] × [5, 6]
 [0.5, 0.600001] × [3, 3.5] × [5, 6]
 [0.5, 0.600001] × [3.6, 4] × [5, 6]
 [0.5, 0.600001] × [3.5, 3.60001] × [5, 5.5]
 [0.5, 0.600001] × [3.5, 3.60001] × [5.6, 6]
 ```